src/main/java/net/blay09/mods/waystones/client/WaystoneCreativeTab.java

```diff
@@ -0,0 +1,21 @@
+package net.blay09.mods.waystones.client;
+
+import net.blay09.mods.waystones.Waystones;
+import net.minecraft.item.ItemStack;
+
+public class WaystoneCreativeTab extends net.minecraft.creativetab.CreativeTabs {
+
+	private static ItemStack itemStack;
+
+	public WaystoneCreativeTab() {
+		super("Waystone");
+	}
+
+	@Override
+	public ItemStack getTabIconItem() {
+		if (itemStack == null) {
+			itemStack = new ItemStack(Waystones.blockWaystone);
```

Comment:
Caching here is unnecessary, getTabIconItem is only ever called once by CreativeTabs itself, which already caches the iconItemStack, so you can just return a new ItemStack right away.